crates/aptos-rosetta/src/types/objects.rs

```diff
@@ -701,6 +578,170 @@ impl Transaction {
+            // Find out if it's the 0th sequence number (creation)
+            for (id, value) in data.data.0.iter() {
+                if id == &sequence_number_identifier() {
+                    if let Ok(U64(0)) = serde_json::from_value::<U64>(value.clone()) {
+                        operations.push(Operation::create_account(
+                            operation_index,
+                            Some(OperationStatusType::Success),
+                            address,
+                            maybe_sender
+                                .map(|inner| *inner.inner())
+                                .unwrap_or(AccountAddress::ONE),
+                        ));
+                        operation_index += 1;
+                    }
+                }
+            }
+        } else if data.typ == coin_store_tag {
+            // Account balance change
+            for (id, value) in data.data.0.iter() {
+                if id == &withdraw_events_identifier() {
+                    serde_json::from_value::<CoinEventId>(value.clone()).unwrap();
+                    if let Ok(event) = serde_json::from_value::<CoinEventId>(value.clone()) {
+                        let withdraw_event = EventKey::new_from_address(
+                            &event.guid.guid.id.addr,
+                            event.guid.guid.id.creation_num.0,
+                        );
+                        if let Some(amount) = get_amount_from_event(events, withdraw_event) {
+                            operations.push(Operation::withdraw(
+                                operation_index,
+                                Some(OperationStatusType::Success),
+                                address,
+                                native_coin(),
+                                amount,
+                            ));
+                            operation_index += 1;
+                        }
+                    }
+                } else if id == &deposit_events_identifier() {
+                    serde_json::from_value::<CoinEventId>(value.clone()).unwrap();
+                    if let Ok(event) = serde_json::from_value::<CoinEventId>(value.clone()) {
+                        let withdraw_event = EventKey::new_from_address(
+                            &event.guid.guid.id.addr,
+                            event.guid.guid.id.creation_num.0,
+                        );
+                        if let Some(amount) = get_amount_from_event(events, withdraw_event) {
+                            operations.push(Operation::deposit(
+                                operation_index,
+                                Some(OperationStatusType::Success),
+                                address,
+                                native_coin(),
+                                amount,
+                            ));
+                            operation_index += 1;
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
So if it's not one of these operations, we don't increment the operation index?

Reply:
If it's not one of these operations, it's not included.  The operation index is required to be from 0 - n and in order continuously.  It's to ensure ordering of the array in JSON I'm assuming.